### PR TITLE
[Don't Merge] add sleef shared lib support.

### PR DESCRIPTION
**Pull request by xuhancn**
Background:
1. This PR make sleef shared/static lib aligned to sleef `SLEEF_STATIC_LIBS` defination.
2. Current code maybe has issue on Windows shared sleef lib, but acturally Windows seems no shared sleef lib.

So, keep this PR open and not merge this PR, if no shared sleef issue.

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

File: caffe2/CMakeLists.txt [hunk 1]
```diff
@@ -1847,6 +1847,9 @@ if(BUILD_TEST)
         if(NOT MSVC)
               target_compile_options(${test_name}_${CPU_CAPABILITY} PRIVATE -Wno-ignored-qualifiers)
         endif(NOT MSVC)
+        if(NOT USE_SYSTEM_SLEEF)
```
Comment (jgong5) [inline caffe2/CMakeLists.txt:1850]: Not familiar with this but where is `USE_SYSTEM_SLEEF` defined?

Comment (xuhancn) [reply caffe2/CMakeLists.txt:1850]: @jgong5 Aligned to here: https://github.com/pytorch/pytorch/blob/dd8a24b8b79bf1882ccad8c28f80b5438634b38c/aten/src/ATen/CMakeLists.txt#L435 . It control use sleef shared/static.

**Review by jgong5 — APPROVED** [review]
I'm fine with the change even though I'm not sure if it can really fix the fbcode issue.

**Comment by xuhancn** [comment]
> I'm fine with the change even though I'm not sure if it can really fix the fbcode issue.

Though we not sure it could fix fb_code, this PR make sleef shared/static lib aligned to sleef `SLEEF_STATIC_LIBS` defination.

**Comment by xuhancn** [comment]
> Sorry, what exactly do you want me to do here?

It seems the SLEEF works well on main branch for a while. 
Shared library support is not necessary.
I will close this PR. Thanks.

**Pull request closed without merging**